website/src/state/poe/generators.ts

```diff
@@ -28,55 +32,26 @@ export const buildShowcase = createSelector(
 );
 
 // orbs
-const buildAlchemy = createSelector(
-  getMods,
-  mods => Alchemy.build(mods),
-);
+const buildAlchemy = createSelector(getMods, mods => Alchemy.build(mods));
 const buildAnnullment = () => new Annulment();
-const buildAugment = createSelector(
-  getMods,
-  mods => Augment.build(mods),
-);
-const buildAlteration = createSelector(
-  getMods,
-  mods => Alteration.build(mods),
-);
-const buildChaos = createSelector(
-  getMods,
-  mods => Chaos.build(mods),
-);
-const buildExalted = createSelector(
-  getMods,
-  mods => Exalted.build(mods),
-);
-const buildRegal = createSelector(
-  getMods,
-  mods => Regal.build(mods),
-);
+const buildAugment = createSelector(getMods, mods => Augment.build(mods));
+const buildAlteration = createSelector(getMods, mods => Alteration.build(mods));
+const buildChaos = createSelector(getMods, mods => Chaos.build(mods));
+const buildExalted = createSelector(getMods, mods => Exalted.build(mods));
+const buildRegal = createSelector(getMods, mods => Regal.build(mods));
 const buildScouring = () => new Scouring();
-const buildTransmute = createSelector(
-  getMods,
-  mods => Transmute.build(mods),
-);
-const buildVaal = createSelector(
-  getMods,
-  mods => Vaal.build(mods),
-);
+const buildTransmute = createSelector(getMods, mods => Transmute.build(mods));
+const buildVaal = createSelector(getMods, mods => Vaal.build(mods));
 // misc
-const buildIncursion = createSelector(
-  getMods,
-  mods => IncursionTempleMods.build(mods),
-);
-const buildElderMods = createSelector(
-  getMods,
-  mods => ElderMods.build(mods),
-);
-const buildShapedMods = createSelector(
-  getMods,
-  mods => ShapedMods.build(mods),
+const buildIncursion = createSelector(getMods, mods =>
+  IncursionTempleMods.build(mods),
 );
+const buildElderMods = createSelector(getMods, mods => ElderMods.build(mods));
+const buildShapedMods = createSelector(getMods, mods => ShapedMods.build(mods));
 
-export function buildGeneratorFactory(generator: string) {
+export function buildGeneratorFactory(
+  generator: string,
+): (state: { poe: PoeState }) => Generator<Mod, Item> {
```

Comment:
In redux-saga the `select` effect can be called with additional arguments for the selector. This requires a single call signature for the selector. Before the selector from buildGeneratorFactory had a union call signature. 

This is technically fine since we didn't provide any additional arguments. Typing this correctly to check the additional type case for a single signature but allowing different for no type arguments sounds very complicated.

Reply:
@dependabot squash and merge